lib/bigcommerce/lightstep/traceable.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module Bigcommerce
+  module Lightstep
+    ##
+    # Module for adding drop-in tracing to any method. Example usage:
+    #
+    # ```ruby
+    # class MyService
+    #   include ::Bigcommerce::Lightstep::Traceable
+    #
+    #   trace :call, 'operation.do-my-thing' do |span, product, options|
+    #     span.set_tag('store_id', request.store_id)
+    #   end
+    #   # or, with no block:
+    #   trace :call, 'operation.do-my-thing'
+    #
+    #   def call(product:, options:)
+    #     # ...
+    #   end
+    # end
+    # ```
+    #
+    module Traceable
+      def self.included(base)
+        base.extend ClassMethods
+      end
+
+      ##
+      # Extend the class with the tracing methods.
+      #
+      module ClassMethods
+        ##
+        # Trace the perform method for the command with the given operation name as the span name
+        #
+        # @param [Symbol] method_name The method to trace
+        # @param [String] operation_name The name to give the span
+        # @param [Hash] tags A key/value hash of tags to set on the created span
+        # @param [Proc] span_block A block to yield before calling perform; useful for setting tags on the outer span
+        #
+        def trace(method_name, operation_name, tags: nil, &span_block)
```

Comment:
I didn't notice a test or docs for the `tags` parameter. Should probably add those to insure functionality.